src/connection.js

```diff
@@ -62,11 +62,14 @@ export default class {
         .send(options.send)
         .end((err, response) => {
           if (err) {
-            let errorString = JSON.parse(err.response.text).err;
-            return reject({
-              code: err.response.status,
-              text: errorString,
-            });
+            if (err.response) {
+              let errorString = JSON.parse(err.response.text).err;
```

Comment:
A side note here, but `response.body` usually has the parsed response text. We could save a step and just use `err.response.body.err` unless there's a special reason not to (I didn't verify this in integration so perhaps it's different in the real world).